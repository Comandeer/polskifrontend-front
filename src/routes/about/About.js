import React from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import styles from './About.styl';
import ResponsivePanel from '../../components/Responsive/ResponsivePanel';
import Link from '../../components/Link/Link';
import HeaderSettings from '../../components/Layout/HeaderSettings';

const About = (props) => {
  return (
    <div className={styles.container}>
      <HeaderSettings description={props.description} title={props.title} currentPath={props.context.path}/>
      <ResponsivePanel header="O serwisie" description="">
        <div className={styles.nav}>
          <Link className={styles.back} to="/">
            <i className="icon-left-big">
            </i>
            Strona główna
          </Link>
        </div>
        <div className={styles.message}>
          <h3>Po co jest ten serwis?</h3>
          <p className={styles.text}>Serwis <strong>Polski Front-End</strong> powstał w celu zebrania w jednym miejscu jak największej liczby stron, serwisów oraz blogów na temat szeroko rozumianego front-end developmentu. Co ważne, wszystkie zgromadzone tutaj serwisy tworzone są <strong>w języku polskim</strong>!</p>
          <p className={styles.text}>Autorem tej strony jestem ja, czyli <strong>Bartek Dybowski</strong>. Oprócz tego serwisu prowadzę również bloga <a href="https://nafrontendzie.pl" target="_blank" title="Na Frontendzie">Na Frontendzie</a>, na którym piszę na temat technologii webowych (a także świata IT ogólnie). Temat ten jest mi więc szczególnie bliski i stąd właśnie pomysł na stworzenie tego serwisu.</p>
          <p className={styles.text}>Podstawowym założeniem "Polskiego Front-Endu" było zebranie tutaj <strong>całych blogów</strong>, a nie pojedynczych artykułów. Z tego powodu selekcji treści mogę dokonywać jedynie na etapie dodawania danej strony do serwisu. Jako, że nie każdy pisze tylko i wyłącznie o front-endzie, to czasem może się tutaj pojawić wpis na temat, na przykład, PHP... Wydaje mi się jednak, że każdy blog czy strona, która pisze <strong>dużo o front-endzie</strong>, czy ogólniej o web developmencie, zasługuje by się tutaj znaleźć... Nawet jeśli front-end to nie jedyny poruszany tam temat.</p>
          <p className={styles.text}>Jeżeli więc znasz (lub sam prowadzisz) jakiś blog, serwis lub stronę w <strong>języku polskim</strong>, której tutaj nie ma, a uważasz, że powinna się tu znaleźć - <strong>zapraszam do zgłoszenia</strong> go przez specjalnie do tego celu <Link to="/zglos-serwis">przygotowany formularz</Link>. Obiecuję, że przejrzę każde zgłoszenie i jeśli uznam, że dana strona się nadaje - dodam ją do serwisu!</p>
          <p className={styles.text}>A jeśli podoba Ci się moja idea i sam prowadzisz bloga o front-endzie, to oprócz zgłoszenia zachęcam również, do <strong>polecania tej strony</strong> swoim czytelnikom! Im większy ruch tutaj <strong>tym większy ruch do Twojego bloga</strong>...</p>
        </div>
      </ResponsivePanel>
    </div>
  );
};

export default withStyles(styles)(About);
