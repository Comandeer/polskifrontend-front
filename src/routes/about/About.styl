@import "../../styles/globals.styl"
@import "../../styles/variables.styl"
@import "../../styles/colors.styl"

.container
  padding 0 $global-padding $global-margin
  background transparent

  @media screen and (min-width: 768px)
    padding 0 0 $global-margin

  .nav
    text-align left
    margin-top $global-margin * 1.5

    .back
      font-weight bold
      display inline-block
      padding $global-padding * 0.75
      background $color-secondary
      color $color-tertiary
      border 3px solid $color-tertiary
      border-radius $global-radius
      text-decoration none
      transition background $animation-time

      @media screen and (min-width: 768px)
        padding $global-padding ($global-padding * 2)

      &:hover
        background $color-tertiary
        color $color-secondary

      &:active
        background $color-secondary

      i
        font-size 20px
        margin-right 10px
        vertical-align top

  .message
    border-left 4px solid $color-tertiary
    padding $global-margin
    margin $global-margin * 1.5 0 $global-margin
    text-align left
    transition border-bottom $animation-time, border-left $animation-time
    position relative
    background $color-secondary
    box-shadow 1px 1px 6px 1px lighten($color-primary, 80%)
    border-radius $global-radius

    h3
      padding 0 0 $global-padding
      text-transform none
      font-size 20px

    .text
      text-align left
      padding $global-padding 0
      line-height 1.6em
      font-size 1.1em
