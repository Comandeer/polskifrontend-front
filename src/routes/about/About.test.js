/* eslint-disable */
import React from 'react';
import About from './About';
import renderer from 'react-test-renderer';
import StyleProvider from '../../../test/StyleProvider';

describe('About component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <StyleProvider insertCss={() => {}}>
          <About context={{ path: '/test' }} />
        </StyleProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
