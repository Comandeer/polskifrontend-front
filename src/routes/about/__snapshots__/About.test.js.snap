// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <h2
        className="container__title"
      >
        <div>
          <span>
            O serwisie
          </span>
        </div>
      </h2>
      <div
        className="wrapper"
      >
        <p
          className="wrapper__description"
        >
          
        </p>
        <div
          className="nav"
        >
          <a
            className="back"
            href="/"
            onClick={[Function]}
          >
            <i
              className="icon-left-big"
            />
            Strona główna
          </a>
        </div>
        <div
          className="message"
        >
          <h3>
            Po co jest ten serwis?
          </h3>
          <p
            className="text"
          >
            Serwis 
            <strong>
              Polski Front-End
            </strong>
             powstał w celu zebrania w jednym miejscu jak największej liczby stron, serwisów oraz blogów na temat szeroko rozumianego front-end developmentu. Co ważne, wszystkie zgromadzone tutaj serwisy tworzone są 
            <strong>
              w języku polskim
            </strong>
            !
          </p>
          <p
            className="text"
          >
            Autorem tej strony jestem ja, czyli 
            <strong>
              Bartek Dybowski
            </strong>
            . Oprócz tego serwisu prowadzę również bloga 
            <a
              href="https://nafrontendzie.pl"
              target="_blank"
              title="Na Frontendzie"
            >
              Na Frontendzie
            </a>
            , na którym piszę na temat technologii webowych (a także świata IT ogólnie). Temat ten jest mi więc szczególnie bliski i stąd właśnie pomysł na stworzenie tego serwisu.
          </p>
          <p
            className="text"
          >
            Podstawowym założeniem "Polskiego Front-Endu" było zebranie tutaj 
            <strong>
              całych blogów
            </strong>
            , a nie pojedynczych artykułów. Z tego powodu selekcji treści mogę dokonywać jedynie na etapie dodawania danej strony do serwisu. Jako, że nie każdy pisze tylko i wyłącznie o front-endzie, to czasem może się tutaj pojawić wpis na temat, na przykład, PHP... Wydaje mi się jednak, że każdy blog czy strona, która pisze 
            <strong>
              dużo o front-endzie
            </strong>
            , czy ogólniej o web developmencie, zasługuje by się tutaj znaleźć... Nawet jeśli front-end to nie jedyny poruszany tam temat.
          </p>
          <p
            className="text"
          >
            Jeżeli więc znasz (lub sam prowadzisz) jakiś blog, serwis lub stronę w 
            <strong>
              języku polskim
            </strong>
            , której tutaj nie ma, a uważasz, że powinna się tu znaleźć - 
            <strong>
              zapraszam do zgłoszenia
            </strong>
             go przez specjalnie do tego celu 
            <a
              href="/zglos-serwis"
              onClick={[Function]}
            >
              przygotowany formularz
            </a>
            . Obiecuję, że przejrzę każde zgłoszenie i jeśli uznam, że dana strona się nadaje - dodam ją do serwisu!
          </p>
          <p
            className="text"
          >
            A jeśli podoba Ci się moja idea i sam prowadzisz bloga o front-endzie, to oprócz zgłoszenia zachęcam również, do 
            <strong>
              polecania tej strony
            </strong>
             swoim czytelnikom! Im większy ruch tutaj 
            <strong>
              tym większy ruch do Twojego bloga
            </strong>
            ...
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
